pkg/perf/monitor.go

```diff
@@ -0,0 +1,100 @@
+package perf
+
+import (
+	"context"
+	"encoding/json"
+	"time"
+
+	"github.com/go-redis/redis"
+	"github.com/jasonlvhit/gocron"
+	"github.com/pkg/errors"
+)
+
+// TaskMethod is the task method signature
+type TaskMethod func(ctx context.Context) (interface{}, error)
+
+type Task struct {
+	Key            string
+	Method         TaskMethod
+	Interval       time.Duration // interval in seconds
+	ExecutionCount uint64
+}
+
+// PerformanceMonitor holds the module data
+type PerformanceMonitor struct {
+	Scheduler   *gocron.Scheduler
+	RedisClient *redis.Client
+	Tasks       []Task
```

Comment:
Why these fields are public? they open the door to bad design since you will end up accessing an `internal` implementation details to an object. A better approach is that all fields become private and then functionality is either defined by interfaces or public methods. 

Some fields can become public of course, but usually in Data only types. Not operational types. For example there is noway someone will ever need to access the redis client on the Monitor structure because that is implementation details of how the Monitor caches the result!